test/integration/namespace/ns_conditions_test.go

```diff
@@ -115,6 +116,35 @@ func TestNamespaceCondition(t *testing.T) {
 	}
 }
 
+// TestNamespaceLabels tests for default labels added in https://github.com/kubernetes/kubernetes/pull/96968
+func TestNamespaceLabels(t *testing.T) {
+	closeFn, _, _, kubeClient, _ := namespaceLifecycleSetup(t)
+	defer closeFn()
+	nsName := "test-namespace-labels-generated"
+	// Create a new namespace w/ no name
+	ns, err := kubeClient.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: nsName,
+		},
+	}, metav1.CreateOptions{})
+
+	if ns.Name != ns.Labels[corev1.LabelMetadataName] {
+		t.Fatal(fmt.Errorf("expected %q, got %q", ns.Name, ns.Labels[corev1.LabelMetadataName]))
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
if err is nil, ns will also be nil and you'll panic on line 131... check this before doing anything with `ns`

Reply:
this is still outstanding

Reply:
fixed, will push